File: Chapter 02/Problems/LinearSearch.py
def linearSearch(arr, key):
    for i in range(len(arr)):
        # key found in arr-> return index
        if arr[i] == key:
            return i
    # key not found in arr-> return -1
    return -1


# main
if __name__== "__main__":
    # array of elements
    arr = list(map(int, input("Enter array elements: ").split()))
    # key to be searched
    key = int(input("Enter key to be searched: "))
    i = linearSearch(arr, key)
    if i == -1:
        print(f"{key} not found")
    else:
        print(f"{key} found at position {i+1} in array")

File: Chapter 02/Problems/LinearSearch.cpp
// Problem 2.1-4
#include<iostream>
#include<vector>
using namespace std;

// linearSearch function
int linearSearch(vector<int> arr,int key){
    for(int i=0;i<arr.size();i++){
        // element found in array => return index
        if (arr[i]==key){
            return i;
        }
    }
    // element not found in array => return -1
    return -1;
}


int main(){
    // size of array
    cout<<"Enter size of array: ";
    int n;cin>>n;
    // array declaration
    vector<int> arr;
    cout<<"Enter the elements of the array:"<<endl;
    // taking array input
    for(int i=0;i<n;i++){
        int x;cin>>x;
        arr.push_back(x);
    }
    // element to search for
    cout<<"Enter the element to search for: ";
    int key;cin>>key;
    int i=linearSearch(arr, key);
    if(i==-1){
        cout<<endl<<key<<" is not present in array"<<endl;
    }
    else{
        cout<<endl<<key<<" is present at position "<<i+1<<" in array"<<endl;
    }
    return 0;
}